src/Pages/PokemonView/PokemonView.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useMemo, useState } from "react";
+import Typography from "../../Components/Typography/Typography";
+import { TypographyTypes } from "../../Components/Typography/consts";
+import HeaderPokemonView from "./components/HeaderPokemonView/HeaderPokemonView";
+import { ViewType } from "./components/HeaderPokemonView/consts";
+import { sortType } from "../../Components/Table/consts";
+import {
+  fetchPokemonData,
+  getCols,
+  getRows,
+  createPokemonCards,
+} from "../dataUtils";
+import Table from "../../Components/Table/Table";
+import { GridEventListener } from "@mui/x-data-grid";
+import { Pokemon } from "./Pokemon";
+import { getTableStyle, MainContainer, TitleContainer } from "./styles";
+import { CardProps } from "../../Components/Card/types";
+import CardsView, { sortCards } from "./components/CradsView/CardsView";
+import { PokemoneViewProps } from "./types";
+
+const PokemonView = ({ title, onPokemonClick }: PokemoneViewProps) => {
+  const [searchBy, setSearchBy] = useState("");
+  const [pokemonCards, setPokemonCards] = useState<CardProps[]>([]);
+  const [clickedPokemon, setClickedPokemon] = useState("");
+  const [sortBy, setSortBy] = useState({ col: "ID", order: sortType.ASC });
+  const [viewOption, setViewOption] = useState<ViewType>(ViewType.TABLE);
+  const [pokemons, setPokemons] = useState<Pokemon[]>([]);
+  const [rows, setRows] = useState<any[]>([]);
+
+  useEffect(() => {
+    const loadData = async () => {
+      const data = await fetchPokemonData();
+      setPokemons(data);
+      setRows(getRows(data));
+      setPokemonCards(createPokemonCards(data, onPokemonClick));
+    };
+
+    loadData();
+  }, []);
+
+  const pokemonMemoCards = useMemo(() => {
+    return createPokemonCards(pokemons, onPokemonClick);
+  }, [pokemonCards]);
+
+  useEffect(() => {
+    const filteredCards =
+      searchBy.trim() === ""
+        ? pokemonMemoCards
+        : pokemonCards.filter((card) =>
+            card.name.toLowerCase().includes(searchBy.toLowerCase())
+          );
+    const sortedCards = sortCards(filteredCards, sortBy);
+
+    setPokemonCards(sortedCards);
+  }, [searchBy, sortBy]);
+
+  const filteredRows = rows.filter((row) => {
+    if (searchBy.trim() === "") {
+      return true;
+    }
+    return row.name.toLowerCase().includes(searchBy.toLowerCase());
+  });
```

Comment:
this is very unefficient.
if there is no search value you still iterste through all of the pokmeons , because you check the search value for each one!

```
  const filteredRows = searchBy.trim() === "" ? rows : rows.filter(row => row.name.toLowerCase().includes(searchBy.toLowerCase()))
```